lang/rust/patches/00-default_dynamic_linking.patch

```diff
@@ -0,0 +1,11 @@
+--- a/compiler/rustc_target/src/spec/linux_musl_base.rs
++++ b/compiler/rustc_target/src/spec/linux_musl_base.rs
+@@ -10,7 +10,7 @@
+     base.crt_objects_fallback = Some(CrtObjectsFallback::Musl);
+
+     // These targets statically link libc by default
+-    base.crt_static_default = true;
++    base.crt_static_default = false;
```

Comment:
cc https://zulip-archive.rust-lang.org/131828tcompiler/75025bootstrapingons390xmipsformusl.html#226006413

Reply:
Ah, So you ARE - in fact - that Joshua :)  Yes, this is why I went this route because rust assumes MUSL as statically linked (and hard-float for MIPS64, actually) which is where I ran into the issues you and I worked on with not being able to use rustup.  In the OpenWrt environment, using MUSL will mean dynamically linked, so I set it to that by default for MUSL